tools/provision.ps1

```diff
@@ -0,0 +1,229 @@
+
+# URL of where our pre-compiled third-party dependenices are archived
+$THIRD_PARTY_ARCHIVE_URL = 'https://s3.amazonaws.com/osquery-pkgs/chocolatey'
+
+# Adapted from http://www.jonathanmedd.net/2014/01/testing-for-admin-privileges-in-powershell.html
+function Test-IsAdmin {
+  return ([Security.Principal.WindowsPrincipal][Security.Principal.WindowsIdentity]::GetCurrent()).IsInRole(
+    [Security.Principal.WindowsBuiltInRole] "Administrator"
+  )
+}
+
+# Attempts to install chocolatey if not already
+function Install-Chocolatey {
+  Write-Host "  Attemping to detect presence of chocolatey..." -foregroundcolor DarkYellow
+  if ((Get-Command 'choco.exe' -ErrorAction SilentlyContinue) -eq $null) {
+    Write-Host "    => Did not find. Installing chocolatey..." -foregroundcolor Cyan
+    iex ((new-object net.webclient).DownloadString('https://chocolatey.org/install.ps1'))
+    $env:Path = "$env:Path;$env:ALLUSERSPROFILE\chocolatey\bin"
+  } else {
+    Write-Host "    => Chocolatey is already installed." -foregroundcolor Green
+  }
+}
+
+# Attempts to install a chocolatey package of a specific version if 
+# not already there.
+function Install-ChocoPackage {
+param(
+  [string] $packageName = '',
+  [string] $packageVersion = '',
+  [array] $packageOptions = @()
+) 
+  Write-Host "  Determining whether $packageName is already installed..." -foregroundcolor DarkYellow
+  
+  $requiresInstall = $false
+  $out = choco list -lr
+  
+  # Parse through the locally installed chocolatey packages and look
+  # to see if the supplied package already exists
+  $found = $false
+  $pktList = $out[1..($out.count - 1)]
+  ForEach ($pkg in $pktList) {
+    $name, $version = $pkg -split '\|'
+    
+    if ($name -eq $packageName) {
+      if ($packageVersion -eq '') {
+        $found = $true
+      } else {
+        if ($version -eq $packageVersion) {
+          $found = $true
+        }
+      }
+    }
+  }
+  
+  if (-not $found) {
+    $requiresInstall = $true
+  }
+  
+  if ($requiresInstall) {
+    Write-Host "    => Did not find. Installing $packageName $packageVersion" -foregroundcolor Cyan
+    
+    $args = @("install", "-y", "${packageName}")
+    if ($packageVersion -ne '') {
+      $args += @("--version", "${packageVersion}")
+    }
+    if ($packageOptions.count -gt 0) {
+      Write-Host "       Options: $packageOptions" -foregroundcolor Cyan
+      $args += ${packageOptions}
+    }
+
+    choco ${args}
+    
+    if ($LastExitCode -ne 0) {
+      Write-Host "      [!] $packageName $packageVersion failed to install!" -foregroundcolor Red
+      Exit -1
+    }
+    
+    Write-Host "      [*] Done." -foregroundcolor Green
+  } else {
+    Write-Host "    => $packageName $packageVersion already installed." -foregroundcolor Green
+  }
+}
+
+function Install-PipPackages {
+  Write-Host "  Attempting to install Python packages" -foregroundcolor DarkYellow
+  
+  $env:Path = "$env:Path;$env:HOMEDRIVE\tools\python2;$env:HOMEDRIVE\tools\python2\Scripts"
```

Comment:
Should this be a prepend of the tools directory? As it stands, if the user already has python installed we'll be using the currently installed version of pip, as opposed to the one we're dropping on the machine.